### PR TITLE
docs: update license file to explicitly use bsd-2 clause license

### LINKED ISSUE #10280: Docs: Update project license to explicitly say BSD-2-Clause
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

•	Npm states that @typescript-eslint/parser uses the BSD-2-Clause license (https://www.npmjs.com/package/@typescript-eslint/parser)
•	The GitHub license has similar text (it appears to be cut-and-pasted from another project) but does not explicitly state BSD-2-Clause (https://github.com/typescript-eslint/typescript-eslint?tab=License-1-ov-file#readme)

Please correct your GitHub license file to explicitly be the BSD-2-Clause to assist with automated pipelines that rely on having an authoritative license for approval.

### Affected URL(s)

https://github.com/typescript-eslint/typescript-eslint?tab=License-1-ov-file#readme

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by peanutenthusiast**
## PR Checklist

- [x] Addresses an existing open issue: fixes #10280 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Updated the license information to the above. Let me know what other edits are needed!

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -3,24 +3,27 @@ typescript-eslint
 Originally extracted from:
 
 TypeScript ESLint Parser
-Copyright JS Foundation and other contributors, https://js.foundation
+BSD 2-Clause License
+
+Copyright (c) 2024, Josh Goldberg ✨
```
Comment (JoshuaKGoldberg) [inline LICENSE:8]: I didn't meant to indicate in https://github.com/typescript-eslint/typescript-eslint/issues/10280#issuecomment-2455579006 that we should switch typescript-eslint's licenses to be under my name! I was just posting that as an reference.

(the idea of an emoji being in the license is fun though)

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for the overhaul, but this is much more work than we need. All this PR should do is add `BSD-2-Clause` on top of the license files.

File: LICENSE (hunk 1 above)
Comment (peanutenthusiast) [reply LICENSE]: Alright, I added the bsd-2-clause to the license files for [typescript-estree](https://www.npmjs.com/package/@typescript-eslint/typescript-estree) and [typescript-eslint/plugin](https://www.npmjs.com/package/@typescript-eslint/parser).

**Review by bradzacher — APPROVED** [review]
thanks for this!

**Pull request merged**